Tools/paletteable sprite editor/js/marvinj-0.9.js

```diff
@@ -0,0 +1,2665 @@
+				cornernessMapOut[x][y] = this.nonmax(x,y,matrixSize,cornernessMap);
+				
+				if(cornernessMapOut[x][y] > 0){
+					cornernessMapOut[x][y] = 1;
+				}
+			}
+		}
+		
+		if(attrOut != null){
+			attrOut.set("cornernessMap", cornernessMapOut);
+		}
+	}
+	
+	Moravec.prototype.nonmax = function(x, y, matrixSize, matrix){
+		var s = Math.floor(matrixSize/2);
+		if(x-(s+1) >= 0 && x+(s+1) < matrix.length && y-(s+1) >= 0 && y+(s+1) < matrix[0].length){
+			for(var i=-s; i<=s; i++){
+				for(var j=-s; j<=s; j++){
+					if(i != 0 || j != 0){
+						if(matrix[x][y] < matrix[x+i][y+j]){
+							return 0;
+						}
+					}
+				}
+			}
+		}
+		return matrix[x][y];
+	}
+	
+	Moravec.directions = [[1,0], [-1, 0], [0, 1], [0,-1], [-1,-1], [1, -1], [-1, 1], [1,1]];
+	
+	Moravec.prototype.c = function(x, y, matrixSize, image){
+		
+		var ret = -1;
+		var temp;
+		var s = Math.floor(matrixSize/2);
+		if(x-(s+1) >= 0 && x+(s+1) < image.getWidth() && y-(s+1) >= 0 && y+(s+1) < image.getHeight()){
+			
+			for(var d=0; d<Moravec.directions.length; d++){
+				temp=0;
+				for(var i=-s; i<=s; i++){
+					for(var j=-s; j<=s; j++){
+						temp += Math.pow(image.getIntComponent0(x+i, y+j)-image.getIntComponent0(x+i+Moravec.directions[d][0],y+j+Moravec.directions[d][1]), 2);
+					}
+				}
+				if(ret == -1 || temp < ret){
+					ret = temp;
+				}
+			}
+			
+		}
+		return ret;
+	}
+
+
+
+
+
+/**
+ * @author Gabriel Ambr�sio Archanjo
+ */
+	function Prewitt(){
+		MarvinAbstractImagePlugin.super(this);
+	
+	
+		// Definitions
+		this.matrixPrewittX = [
+				[1,		0,		-1],
+				[1,		0,		-1],
+				[1,		0,		-1]
+		];
+		
+		this.matrixPrewittY = [
+				[1,		1,		1],
+				[0,		0,		0],
+				[-1,	-1,		-1]
+		];
+		
+		this.load();
+		
+	}
+	
+	Prewitt.prototype.load = function(){
+		this.convolution = new Convolution();
+		this.setAttribute("intensity", 1.0);
+	};
+	
+	Prewitt.prototype.process = function
+	(
+		imageIn, 
+		imageOut,
+		attrOut,
+		mask, 
+		previewMode
+	)
+    {
+		var intensity = this.getAttribute("intensity");
+		
+		if(intensity == 1){
+			this.convolution.setAttribute("matrix", this.matrixPrewittX);
+			this.convolution.process(imageIn, imageOut, null, mask, this.previewMode);
+			this.convolution.setAttribute("matrix", this.matrixPrewittY);
+			this.convolution.process(imageIn, imageOut, null, mask, this.previewMode);
+		} else{
+			this.convolution.setAttribute("matrix", MarvinMath.scaleMatrix(this.matrixPrewittX, intensity));
+			this.convolution.process(imageIn, imageOut, null, mask, previewMode);
+			this.convolution.setAttribute("matrix", MarvinMath.scaleMatrix(this.matrixPrewittY, intensity));
+			this.convolution.process(imageIn, imageOut, null, mask, previewMode);
+		}
+    };
+
+	function BoundaryFill(){
+		MarvinAbstractImagePlugin.super(this);
+		this.load();
+	}
+	
+	BoundaryFill.prototype.load = function(){
+		this.setAttribute("x", 0);
+		this.setAttribute("y", 0);
+		this.setAttribute("color", 0xFFFF0000);
```

Comment:
Codacy found an issue: [The numeric literal '0xFFFF0000' will have at different value at runtime.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6395467)